### PR TITLE
feat(sdk)!: add delete method for MutJson class

### PR DESCRIPTION
**Pull request by partha04patel**
## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

BREAKING CHANGE: `has()` removed from static `Json`. It is now an instance method on `Json` objects.

### CODE REVIEW

**Comment by partha04patel** [comment]
meant as resolution for https://github.com/winglang/wing/issues/5916

File: examples/tests/sdk_tests/std/json.test.w [hunk 1]
```diff
@@ -178,4 +178,25 @@ test "deepCopy(), deepCopyMut()" {
   assert(copy != copyMut);
 
   assert(copyMut.get("object") == mutation);
+}
+
+test "delete() for MutJson" {
+  let mutObj = MutJson { x: 1, y: 2 };
+  mutObj.delete("x");
+  let assertThrows = (expected: str, block: (): void) => {
```
Comment (eladb) [inline examples/tests/sdk_tests/std/json.test.w:186]: I think you can just use `Json.has(x)` instead to verify that the key was deleted. No?

Comment (partha04patel) [reply examples/tests/sdk_tests/std/json.test.w:186]: I don't think I can since the has() method doesn't exist for the MutJson class from what I understand

Comment (partha04patel) [reply examples/tests/sdk_tests/std/json.test.w]: has() method added to MutJson class as part of this PR!

**Review by MarkMcCulloh — APPROVED** [review]

File: libs/wingsdk/src/std/json.ts [hunk 2]
```diff
@@ -453,4 +449,31 @@ export class MutJson {
...
+   * @macro (delete ($self$)[$args$])
+   *
+   * @param key The key
+   * @returns true if the given key is no longer present
+   */
+
+  public delete(key: string): boolean {
+    key;
+    throw new Error("Macro");
+  }
```
Comment (Chriscbr) [inline libs/wingsdk/src/std/json.ts:465]: What happens if `delete()` is called on something that isn't a map? like `Json true` or `Json [1, 2, 3]`? Can we add tests for this?

Comment (MarkMcCulloh) [reply libs/wingsdk/src/std/json.ts:465]: It should return true (matching the return comment). 
null is also technically valid json and that would throw here (bad), but most of our json methods wouldn't correctly handle null anyways.
Agree a test would be good, my b for approving without more tests.

Comment (partha04patel) [reply libs/wingsdk/src/std/json.ts:465]: I could try adding extra tests for some of those edge cases in another PR!
